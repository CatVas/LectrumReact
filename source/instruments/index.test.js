
import { delay, getUniqueID, sum } from './';

describe('instruments:', () => {
    test('sum should be a function', () => {
        expect(sum).toBeInstanceOf(Function);
    });

    test('sum should throw with non-number 1st arg', () => {
        expect(() => sum('2', 2)).toThrow();
    });

    test('sum should throw with non-number 2nd arg', () => {
        expect(() => sum(2, '')).toThrow();
    });

    test('sum should return addition', () => {
        expect(sum(2, 2)).toBe(2 + 2);
        expect(sum(4, 7)).toMatchSnapshot();
    });

    test('delay should return a resolved promise', async () => {
        await expect(delay()).resolves.toBeUndefined();
    });

    test('getUniqueID should be a function', () => {
        expect(getUniqueID).toBeInstanceOf(Function);
    });

    test('getUniqueID throws with non-number arg', () => {
        expect(() => getUniqueID('20')).toThrow();
    });

    test('getUniqueID returns a string of desired length', () => {
        const res = getUniqueID(20);
        expect(typeof res).toBe('string');
        expect(res).toHaveLength(20);
    });
});
