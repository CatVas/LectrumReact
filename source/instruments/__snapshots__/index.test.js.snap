// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instruments: sum should return addition 1`] = `11`;
